src/library/dao/macrodao.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+
+#include "library/dao/dao.h"
+#include "track/macro.h"
+#include "track/trackid.h"
+
+class MacroDAO : public virtual DAO {
+  public:
+    bool saveMacro(TrackId trackId, Macro* macro, int slot) const;
+    void saveMacros(TrackId trackId, QMap<int, MacroPointer> macros) const;
```

Comment:
MacroPointer is not declared in scope.

Also:
```suggestion
    void saveMacros(TrackId trackId, const QMap<int, MacroPointer>& macros) const;
```

Reply:
but MacroPointer is declared in `macro.h` which is imported?